### PR TITLE
cities API

### PR DESCRIPTION
**Pull request by Pawlllosss**
Storying city as an entity instead of an enum

### CODE REVIEW

File: src/main/java/pl/orki/hackathon/webapp/band/control/BandService.java [hunk 1]
```diff
@@ -3,20 +3,37 @@
...
-    public BandService(BandRepository bandRepository, UserRepository userRepository) {
+    public BandService(BandRepository bandRepository, UserRepository userRepository, CityRepository cityRepository) {
         this.bandRepository = bandRepository;
         this.userRepository = userRepository;
+        this.cityRepository = cityRepository;
+    }
+
+    public List<Band> getBandsByMusicGenresAndCities(List<String> genresNames, List<Long> citiesIds) {
+        List<City> cities = cityRepository.findAllById(citiesIds);
+        // TODO: it will be replaced by querying genres by ids
```
Comment (JakubKowalski1997) [inline src/main/java/pl/orki/hackathon/webapp/band/control/BandService.java:31]: TODOs lasts forever :P

File: src/main/java/pl/orki/hackathon/webapp/InitialDataProvider.java [hunk 2]
```diff
@@ -3,66 +3,89 @@
...
+    private final Long cityGdanskId;
 
-    public InitialDataProvider(VenueRepository venueRepository, BandRepository bandRepository) {
+    public InitialDataProvider(VenueRepository venueRepository, BandRepository bandRepository, CityRepository cityRepository) {
         this.venueRepository = venueRepository;
         this.bandRepository = bandRepository;
+        this.cityRepository = cityRepository;
+        this.cityKrakowId = createCity("Kraków");
+        this.cityWroclawId = createCity("Wrocław");
+        this.cityGdanskId = createCity("Gdańsk");
```
Comment (JakubKowalski1997) [inline src/main/java/pl/orki/hackathon/webapp/InitialDataProvider.java]: extract to method

File: src/main/java/pl/orki/hackathon/webapp/InitialDataProvider.java [hunk 3]
```diff
@@ -3,66 +3,89 @@
...
     }
 
     @Override
+    @Transactional
     public void run(ApplicationArguments args) throws Exception {
         createVenues();
```
Comment (JakubKowalski1997) [inline src/main/java/pl/orki/hackathon/webapp/InitialDataProvider.java:37]: and add cities here

**Review by JakubKowalski1997 — APPROVED** [review]

File: src/main/java/pl/orki/hackathon/webapp/band/control/BandService.java (hunk 1 above)
Comment (Pawlllosss) [reply src/main/java/pl/orki/hackathon/webapp/band/control/BandService.java:31]: Not this one, after adding tests to these changes I will rewrite Genry to entity from enum as well ;)

**Pull request merged**